ui/src/app-bar/app-bar.tsx

```diff
@@ -30,6 +32,17 @@ const links: Link[] = [
 export const AppBar: React.FC = () => {
     const { t } = useTranslation();
     const [dataMenuOpen, setDataMenuOpen] = React.useState(false);
+    const selectedInfraModelTab = useInfraModelAppSelector((state) => state.infraModelActiveTab);
+
+    function getInfraModelLink(): string {
+        if (selectedInfraModelTab === InfraModelTabType.PLAN) {
```

Comment:
Tästä tuli mieleen (ei tarvitse nyt korjata koska vähän yleisempi juttu) että meidän kannattaisi keksiä tapa tehdä typescriptillä tyyppiturvalliset exhaustive switch-rakenteet. Tämäkin voisi olla switch, mutta nykyisellään se ei juurikaan muuta. Noita ei siis nyt ole mutta esim jollain tällaisella voisi auttaa välttämään virheitä:
https://medium.com/technogise/type-safe-and-exhaustive-switch-statements-aka-pattern-matching-in-typescript-e3febd433a7a

Toinen vaihtoehto (tai lisäksi) olisi laittaa ts optioihin noImplicitReturn päälle ja tehdä if-else rakenteet niin että viimeistä elseä ei ole. Silloin tulee käännösvirhe jos kaikki tapaukset ei ole käsitelty. Muutaman koodipaikan joutuu korjaamaan mutta voisi olla ihan hyvä optio pitää päällä muutenkin.